File: 장태원 node/src/cmd.py
import json

def getData():
  f = open("./data/memo.json", "r", encoding="utf-8")
  return json.load(f)

def add(a, b):
  data = getData()
  result = a + b
  data["list"].append(result)
  print(data, result)
  f = open("./data/memo.json", "w", encoding="utf-8")
  json.dump(data, f, ensure_ascii=False)

def list():
  print("list() 호출됨")
  data = getData()
  arr = data["list"]
  for v in arr: print(v)

def delete():
  data = getData()


def update():
  sdf

File: 이현서 node/src/cmd.js
import fs from 'fs';

const getData = () => {
    const f = fs.readFileSync('./data/memo.json', 'utf-8')
    return JSON.parse(f)
}
const writeData = (data) => {
    fs.writeFileSync('./data/memo.json', JSON.stringify(data), 'utf-8')
}

export const submit = (word) => {
    const data =  getData();   
    data.list.push({id: Date.now(), name: word})
    console.log(data);
    writeData(data);
}

export const list = () => {
    console.log("list() 호출됨");
    const arr = getData()
    for(const v of arr.list) console.log(v);
}

export const Ddelete = (name) => {
    const data = getData()
    data.list = data.list.filter(v => v.name !== name)    
    writeData(data);
    console.log(data)
}

export const update = (targetName, newName) => {
  const data = getData();
  const target = data.list.find(v => v.name === targetName);
  if (!target) {console.log("단어 없음"); return; }
   if (newName) target.name = newName;
  writeData(data);
  console.log(data);
};
